Fix bug when asserting some valid ES6 keys

meeber opened the pull request:
- Resolution of #674
- Add compareByInspect utility for use with assertKeys sorts
- Add getOwnEnumerableProperties utility
- Add getOwnEnumerablePropertySymbols utility
- Add Symbol support to the inspect utility
- Add tests to utilities, should, expect, and assert

keithamus commented on lib/chai/core/assertions.js:
```diff
@@ -1172,6 +1172,11 @@ module.exports = function (chai, _) {
 
+      if (typeof Object.getOwnPropertySymbols === 'function') {
+        // Object.keys excludes Symbols so we need this too
+        actual = actual.concat(Object.getOwnPropertySymbols(obj));
```
Slight concern with this - `Object.getOwnPropertySymbols` returns enumerable _and_ non-enumerable keys - somewhat changing the semantics of what this assertion already does. Should we filter out the non-enumerable keys? Something like:


```js
actual = actual.concat(Object.getOwnPropertySymbols(obj).filter(function (symbol) {
  return Object.getOwnPropertyDescriptor(obj, symbol).enumerable;
}));
````

meeber replied:
Whoops! Great catch. I'm on it :)

meeber commented:
Whoops! Great catch. I'm on it :)

meeber commented:
Fixed the issue with the non-enumerable symbols, and also modified a couple of the keys tests to validate that non-enumerable strings and symbols aren't included.

lucasfcosta commented on lib/chai/utils/getOwnEnumerablePropertySymbols.js:
```diff
@@ -0,0 +1,27 @@
+ *
+ * This allows the retrieval of directly-owned enumerable property symbols of an
+ * object. This function is necessary because Object.getOwnPropertySymbols
+ * returns both enumerable and non-enumerable property symbols.
```
This description seems wrong since this function returns only enumerable symbol properties and not all of them (enumerable and non-enumerable)

lucasfcosta commented:
I just went through it all and it LGTM!
Awesome work, thanks for including complete tests for the features you've just implemented.
I really, really liked it.

meeber commented:
Great, thank you! Just noticed a typo in the comments; gonna fix it real quick.

keithamus commented:
LGTM 👍
